libcircle.spec

```diff
@@ -0,0 +1,236 @@
+%endif
+
+%description openmpi3-devel
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains development headers and libraries for Open
+MPI ibcircle
+%endif
+
+%if %{with_mpich}
+%package mpich
+Summary:        Libcircle MPICH libraries
+BuildRequires:  mpich-devel
+
+%description mpich
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+libcircle compiled with MPICH
+
+%if (0%{?suse_version} >= 1500)
+%package -n libcircle2-mpich
+Summary: Light-weight Group Library for MPI process groups -- Shared libraries
+
+%description -n libcircle2-mpich
+Shared libraries for %{name}-mpich.
+%endif
+
+%package mpich-devel
+Summary:    Development headers and libraries for MPICH libcircle
+%if (0%{?suse_version} >= 1500)
+Requires: libcircle2-mpich%{_isa} = %version-%release
+%else
+Requires:   %{name}-mpich%{?_isa} = %{version}-%{release}
+%endif
+
+%description mpich-devel
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains development headers and libraries for
+MPICH libcircle
+%endif
+
+%prep
+%setup -q
+
+%build
+%global _configure ../configure
+export CC=mpicc
+for mpi in %{?mpi_list}; do
+  mkdir $mpi
+  pushd $mpi
+  %module_load $mpi
+  %configure --enable-doxygen --enable-tests --disable-static --includedir=%{mpi_includedir}/$mpi%{mpi_include_ext} --libdir=%{mpi_libdir}/$mpi/%{mpi_lib_ext}
+  %make_build
+  module purge
+  popd
+done
+
+%install
+for mpi in %{?mpi_list}; do
+  %module_load $mpi
+  %make_install -C $mpi
+  rm %{buildroot}/%{mpi_libdir}/$mpi/%{mpi_lib_ext}/*.la
+done
+
+%check
+for mpi in %{?mpi_list}; do
+  %module_load $mpi
+  %make_install -C $mpi
+  if ! make -C $mpi check; then
+    cat $mpi/tests/test-suite.log
+    exit 1
+  fi
+  rm %{buildroot}/%{mpi_libdir}/$mpi/%{mpi_lib_ext}/*.la
+done
+
+%if %{with_openmpi3}
+%files openmpi3
+%license COPYING
+%doc AUTHORS
+%if (0%{?suse_version} >= 1500)
+%files -n libcircle2-openmpi3
+%endif
+%{mpi_libdir}/openmpi3/%{mpi_lib_ext}/%{name}.so.*
+
+%files openmpi3-devel
+%{mpi_includedir}/openmpi3%{mpi_include_ext}/%{name}.h
+%{mpi_libdir}/openmpi3/%{mpi_lib_ext}/%{name}.so
+%{mpi_libdir}/openmpi3/%{mpi_lib_ext}/pkgconfig/%{name}.pc
+%endif
+
+%if %{with_mpich}
+%files mpich
+%license COPYING
+%doc AUTHORS
+%if (0%{?suse_version} >= 1500)
+%files -n libcircle2-mpich
+%endif
+%{mpi_libdir}/mpich/%{mpi_lib_ext}/%{name}.so.*
+
+%files mpich-devel
+%{mpi_includedir}/mpich%{mpi_include_ext}/%{name}.h
+%{mpi_libdir}/mpich/%{mpi_lib_ext}/%{name}.so
+%{mpi_libdir}/mpich/%{mpi_lib_ext}/pkgconfig/%{name}.pc
+%endif
+
+%changelog
+* Tue Sep 29 2020 Brian J. Murrell <brian.murrell@intel.com> - 0.3.0-2
+- Package for multiple MPI stacks and multiple distros
+
+* Mon Sep 21 2020 John E. Malmberg <john.e.malmberg@intel.com> - 0.3.0-1
+- Use openmpi3 instead of openmpi, add environment-modules depend.
```

Comment:
Merge these into 0.3.0-1 since my change never was completed?

Reply:
I think it's fine as is and gives due credit to where this work started.